src/domain/Customer.ts

```diff
@@ -0,0 +1,28 @@
+import NotEligibleForMortgageException from "../exceptions/NotEligibleForMortgageException";
+
+export default class Customer {
+    constructor(private readonly id: number, public firstName: string, public lastName:string , private balance: number, private badCreditHistoryCount: number) {
+        this.id = id;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.balance = balance;
+        this.badCreditHistoryCount = badCreditHistoryCount;
+    }
```

Comment:
We can simplify this part

```suggestion
    constructor(
        public id: number,
        public firstName: string,
        public lastName: string,
        public balance: number,
        public badCreditHistoryCount: number
    ) {}
```